### ISSUE TITLE
select multiple options on API version

### ISSUE THREAD

**Issue by davideike** [eduardotkoller/convForm#37]
Hi Eduardo - I sent you a $ donation because I love this plugin. I made some changes to hide the text input field when not required and other small things. However, I'm having trouble adding multiple select option functionality to the "API" version. I've been playing with the newState() function, but have not been able to get multiple select options to work yet. Do you have any tips?

Thanks!
David

**Comment by eduardotkoller** [comment]
Hi David, I am really glad you liked the plugin, thanks for the donation, I really appreciate it.

I tested and there really were 2 unrelated bugs with multiple selects:
1. the selected array was not initiated properly so .push() was not working when clicking an answer in multiple selects while using callbacks to build the next state (like when using an API);
2. multiple selects were not being submitted properly when the required attribute was being used.

I fixed those 2 and tested, now it seems to work properly. To use multiple selects, just add `multiple: true` to the newState options object (provided that the object has `type: 'select'` too).

Let me know if it works for you now.
